core/src/main/scala/weaponregex/parser/ParserJVM.scala

```diff
@@ -89,7 +89,7 @@ class ParserJVM private[parser] (pattern: String) extends Parser(pattern) {
     *   `"[abc]"`
     */
   override def charClass[A: P]: P[CharacterClass] =
-    Indexed("[" ~ "^".!.? ~ (charClassIntersection.rep(exactly = 1) | classItem.rep(minCharClassItem)) ~ "]")
+    Indexed("[" ~ "^".!.? ~ ((charClassIntersection.rep(exactly = 1): P[Seq[RegexTree]]) | classItem.rep(minCharClassItem)) ~ "]")
```

Comment:
May I ask why this particular `rep` needs an upcast, but not others?

Reply:
Hello, thank you for having a look at the PR!

In lampepfl/dotty#17924, I made a change in the compiler that affects program elaboration (ie, implicits resolution). The consequence is that the compiler may give up earlier than what it used to do in the past. Here, the type ascription helps the compiler figure out the least upper bound of both sides of the `|` operator, which in turn simplifies the resolution of implicit parameter for the second `rep` call.

Reply:
Thanks for the explanation! This `rep` comes from fastparse so I would imagine there’re more projects that depend on fastparse that are affected by this problem. Should this issue be elevated to fastparse?

Reply:
I am not sure what would be a good fix in fastparse, though…